src/GraphQL/Resolvers/MemberResolver.cs

```diff
@@ -0,0 +1,168 @@
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace GraphQL.Resolvers
+{
+    /// <summary>
+    /// A precompiled field resolver for a specific <see cref="MethodInfo"/>, <see cref="PropertyInfo"/> or <see cref="FieldInfo"/>.
+    /// Returns the specified field or property, or for methods, calls the specified method (with the specified arguments)
+    /// and returns the value of the method.
+    /// </summary>
+    internal class MemberResolver : IFieldResolver
```

Comment:
Mostly this class is a renamed `MethodResolver` with added constructors for `FieldInfo` and `PropertyInfo`.  There will be significant code deduplication with this design once ValueTask is implemented in PR #2907 .